org.osgi.test.common/src/main/java/org/osgi/test/common/inject/FieldInjector.java

```diff
@@ -208,4 +213,26 @@ static boolean isFieldShadowedByLocalFields(Field field, List<Field> localFields
 			.collect(toCollection(ArrayList::new));
 	}
 
+	public static void assertParameterIsOfType(Class<?> actual, Class<?> expected,
+		Class<? extends Annotation> annotationType, Function<String, ? extends RuntimeException> exception) {
+		if (actual != expected) {
+			throw exception.apply("Can only resolve @" + annotationType.getSimpleName()
+				+ " parameter of type " + expected.getName() + " but was: " + actual.getName());
+		}
+	}
+
+	public static void assertFieldIsOfType(Field field, Class<?> expected, Class<? extends Annotation> annotationType,
+		Function<String, ? extends RuntimeException> exception) {
+		if (field.getType() != expected) {
+			throw exception.apply(
+				"[" + field.getName() + "] Can only resolve @" + annotationType.getSimpleName() + " field of type "
+					+ expected.getName() + " but was: " + field.getType()
+						.getName());
+		}
+		if (Modifier.isFinal(field.getModifiers()) || Modifier.isPrivate(field.getModifiers())) {
+			throw exception.apply(
+				'@' + annotationType.getSimpleName() + " field [" + field.getName() + "] must not be private or final");
```

Comment:
`"@"` since we are concatenating strings, we should use String constants. Not a char constant which must be converted to a String object at runtime.

Reply:
Fixed.